test/token-sweep.test.ts

```diff
@@ -0,0 +1,197 @@
+// Start - Support direct Mocha run & debug
+import 'hardhat'
+import '@nomiclabs/hardhat-ethers'
+// End - Support direct Mocha run & debug
+
+import chai, {expect} from 'chai'
+import {before} from 'mocha'
+import {solidity} from 'ethereum-waffle'
+import {
+    SweepERC20TokensHarness,
+    SweepERC721TokensHarness,
+    ERC721PresetMinterPauserAutoId,
+    ERC20PresetMinterPauser
+} from '../typechain'
+import {
+    deployContract,
+    deployContractWithProxy,
+    signer
+} from './framework/contracts'
+import {constants, ContractReceipt, Wallet} from 'ethers'
+import {verifyERC20TransferEventLogs} from './contracts/common/verify-erc20-transfer'
+import {successfulTransaction} from './framework/transaction'
+import {verifyERC721TransferEventLogs} from './contracts/common/verify-erc721-transfer'
+
+// Wires up Waffle with Chai
+chai.use(solidity)
+
+const ADDRESS_ZERO = constants.AddressZero
+
+describe('Token Sweep contracts', () => {
+    let beneficary: string
```

Comment:
question: I've always come across variables being put at the bottom of the tests rather then the top. Have you any good resources debating this, it's a thing we need to keep standard.

Reply:
Have no strong preference. Just mutable variables at the bottom?

Reply:
Yeah, I'll trawl the interwebs for the material ...as I recall it made sense, but placing them at the bottom does mean we can't use the eslint rule about where to place variables 🤔

Reply:
moved!